src/core/reference/CMakeLists.txt

```diff
@@ -21,6 +21,19 @@ add_library(${TARGET_NAME} STATIC ${LIBRARY_SRC} ${PUBLIC_HEADERS})
 add_library(openvino::reference ALIAS ${TARGET_NAME})
 set_target_properties(${TARGET_NAME} PROPERTIES EXPORT_NAME reference)
 
+if(ENABLE_AVX2)
+    ov_avx2_optimization_flags(avx2_flags)
+    string(REPLACE ";" " " avx2_flags "${avx2_flags}")
+
+    set(OV_REFERENCE_X86_AVX2_SRC
+        ${CMAKE_CURRENT_SOURCE_DIR}/src/op/convert_x86_intrinsics.cpp
+    )
+    set_source_files_properties(${OV_REFERENCE_X86_AVX2_SRC} PROPERTIES COMPILE_FLAGS ${avx2_flags}
```

Comment:
I think if you use COMPILE_PTIONS instead of COMPILE_FLAGS, you would not need replace ; with spaces